mlir/lib/Gradient/Transforms/BufferizationPatterns.cpp

```diff
@@ -15,18 +15,46 @@
 #include "iostream"
 #include "llvm/Support/raw_ostream.h"
 
+#include "mlir/Dialect/Index/IR/IndexOps.h"
 #include "mlir/Dialect/MemRef/IR/MemRef.h"
 #include "mlir/Dialect/Tensor/IR/Tensor.h"
 #include "mlir/Transforms/DialectConversion.h"
 
 #include "Gradient/IR/GradientOps.h"
 #include "Gradient/Transforms/Passes.h"
+#include "Gradient/Utils/GradientShape.h"
 
 using namespace mlir;
 using namespace catalyst::gradient;
 
 namespace {
 
+/// Helper function to generate a set of memref allocations.
+///
+/// The allocation size and shape is deduced from a list of existing memref values.
+///
+void generateAllocations(PatternRewriter &rewriter, Location loc,
+                         SmallVectorImpl<Value> &allocations, ValueRange referenceValues)
+{
+    for (Value memref : referenceValues) {
+        MemRefType memrefType = memref.getType().cast<MemRefType>();
```

Comment:
nit: this style of casting is going to be removed
```suggestion
        MemRefType memrefType = cast<MemRefType>(memref.getType());
```

Reply:
Good point, thanks!